controllers/datadogagent/feature/apm/feature.go

```diff
@@ -125,6 +133,11 @@ func (f *apmFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.Requ
 				},
 			},
 		}
+
+		if f.shouldEnableLanguageDetection() {
+			reqComp.Agent.Containers = append(reqComp.Agent.Containers, apicommonv1.ProcessAgentContainerName)
```

Comment:
I think we can use the feature without the process-agent

Reply:
Refer to [this](https://github.com/DataDog/datadog-operator/pull/1195#discussion_r1613558536)

I think no change is needed.

Reply:
Alright but here we mark the container as required. If it becomes possible to collect processes without the process-agent, IMO we should have an option to do it without requiring the component.

Reply:
Makes sense @AliDatadog 
I updated the PR to distinguish between the two cases.

Reply:
👋 With `DD_PROCESS_CONFIG_RUN_IN_CORE_AGENT_ENABLED` set to true, it is indeed possible to run the process check in the core agent _without the process agent_. So in this case the container should not be required, in fact it may result in CLB because the process agent will just quit immediately.

Reply:
It seems like this still needs updating. The process agent container should not be required if the checks are on the core agent. Unless you mean to do:
```go
if !f.processCheckRunsInCoreAgent && f.shouldEnableLanguageDetection() {}
```

Also, if you do want to run language detection on the core agent solely, there should be a check for the process check being enabled.

Reply:
> Unless you mean to do 

Indeed, this is what I intended to do. Will update it.

> Also, if you do want to run language detection on the core agent solely, there should be a check for the process check being enabled.

Not sure about this. My understanding is that at the end we should be able to run language detection in the core agent regardless of whether it is coming from the process check (which requires enabling live process collection) or from the process collector (which should not ideally require live process collection to be enabled).

Reply:
Yes, I agree that should happen in the end. But, I'm not sure what agent version will be default here and I think the Operator can lag behind the latest release. Thus, there's a risk that the process agent won't be required and live processes isn't enabled resulting in language detection not working when expected. A possible alternative here is to have a version check for when we expect for it to work on the core agent (7.57.0 seems most likely).

Reply:
Also, is SSI enabled by default? Seems like no, but is there plans for that to be the case in the near future?

Reply:
> Also, is SSI enabled by default? Seems like no, but is there plans for that to be the case in the near future?

I think the answer is no. 

But maybe #apm-onboarding can have a more decisive answer.

Reply:
We discussed off Github and decided that we would require the process agent for now and update this when equivalent support on the core agent is available.

Reply:
Does this depend on the process agent running any other features? For example, what would happen if all the process agent-based features were disabled, i.e. process discovery disabled, live containers disabled, etc. and language detection was enabled? Would the process agent crash?